### PR TITLE
feat: Technically upgrade to Mangrove v2

### PR DESCRIPTION
**Pull request by espendk**
This PR is mainly a technical upgrade from Mangrove v1 to v2: It fixes configuration, scripts, and paths to work with the MGV v2 repo structures.

In slightly more detail, this PR does the following:

- Updates mangrove-core the latest pre-release version (2.0.0-3)
- Changes strat references to the new mangrove-strats repo and the latest pre-release version (0.0.2-2)
- mangrove.js still needs to be updated, there's no MGV v2 pre-release yet
- Fixes natspec generation and references to work with the new repo structure and remappings
- No rewriting of text has been done
- I took the opportunity to replace verbatim code with references to files as this makes maintenance easier

# Notes
- Some .md files contain verbatim code. We should change these such that they instead reference code in mangrove-{core,strats} or mangrove.js to ease maintenance.
	- Examples:
		- `docs/developers/strat-lib/guides/DirectHowTo.md`
			- I changed this to list `OfferMaker` and `Amplifier` code from mangrove-strats instead
and determine if they should stay.
		- `docs/developers/strat-lib/guides/HowToDeploy.md`
			- I changed this to list code from mangrove-strats instead
		- `docs/developers/strat-lib/guides/HowToTest.md`
			- I changed this to list code from mangrove-strats instead
			- NB: The `Amplifier.t.sol` test splits the test functions in a weird way, where some of the test body is in a separate function.
  - NB: I only updated the code listings to references
    - I did not update the text
    - Some of the referenced code includes some comments that weren't in the verbatim code. We should review these 
- Snipe -> Clean needs rewriting in multiple place
	- `MgvCleaner` has been removed. There are some references to it that should be replaced by references to `Mangrove.cleanByImpersonation`
- I've inserted some FIXMEs when I encountered something that required writing prose.
- Natspec:
  - `IMangrove` is now natspec'ed, so we should include it in the docs.
  - The directory structure for strat-lib natspec is annoying: `code/strats/src/strategies`
    1. It's weird that we have the `strategies` folder in the mangrove-strats repo
    2. The generated docs could be placed directly in `code/`

### CODE REVIEW

File: docs/developers/keeper-bots/background/the-role-of-cleaning-bots-in-mangrove.md
Comment (dontrolle) [inline docs/developers/keeper-bots/background/the-role-of-cleaning-bots-in-mangrove.md:1]: I guess there are some references to MgvCleaner here, which should also go to Mangrove.cleanByImpersonation now? (Just checking, I don't think this needs to be resolved before merging to `develop`)

File: docs/developers/strat-lib/technical-references/code/strats/src/strategies/interfaces/IOrderLogic.md
Comment (dontrolle) [inline docs/developers/strat-lib/technical-references/code/strats/src/strategies/interfaces/IOrderLogic.md:1]: Weird that Git's renaming logic couldn't figure out the renaming here. I guess, Git history on these generated files is not worth conserving? If not - then disregard this comment.

**Review by dontrolle — APPROVED** [review]
Looks good to me. 

Comments are for consideration, not necessarily for fixing now.

File: docs/developers/keeper-bots/background/the-role-of-cleaning-bots-in-mangrove.md
Comment (espendk) [reply docs/developers/keeper-bots/background/the-role-of-cleaning-bots-in-mangrove.md:1]: Indeed. I somehow managed to delete my note on that from the PR description 🤦‍♂️ Re-added

File: docs/developers/strat-lib/technical-references/code/strats/src/strategies/interfaces/IOrderLogic.md
Comment (espendk) [reply docs/developers/strat-lib/technical-references/code/strats/src/strategies/interfaces/IOrderLogic.md:1]: Yeah, I'm not sure why it didn't detect this. But I agree that this is not important for generated files.

**Pull request merged**